SCClassLibrary/Common/Collections/linux/extString_linux.sc

```diff
@@ -1,6 +1,26 @@
 + String {
-	runInTerminal {|shell="sh"|
-		("xterm -hold -T 'SuperCollider runInTerminal' -e" + shell + "-c" + this.shellQuote).unixCmd;
+
+	runInTerminal { |shell="/bin/sh"|
+		LinuxPlatform.runInTerminalCmd ?? { LinuxPlatform.runInTerminalCmd = LinuxPlatform.findTerminalEmulator };
+		if(LinuxPlatform.runInTerminalCmd.isNil) {
+			"String:runInTerminal couldn't find a supported terminal emulator. Please set LinuxPlatform.runInTerminalCmd, see Help for details".error;
+			^ this;
+		} {
+			var fpath = PathName.tmp +/+ "SCStringForTerminal%.sh".format(this.hash.asString);
+			var file = File.open(fpath, "w");
+			if(file.isOpen.not) {"String:runInTerminal was unable to create the script file".error} {
+			fpath = fpath.shellQuote;
+				file.write([
+					"#!" ++ shell,
+					this,
+					"rm" + fpath,
+					"read -p 'Press ENTER to close this window.'"
+				].join($\n));
+				file.close;
+				("/bin/chmod +x" + fpath).systemCmd;
+				LinuxPlatform.runInTerminalCmd.format("SuperCollider in Terminal".shellQuote, fpath).unixCmd
```

Comment:
keep as runInTerminal -- programming is much simpler when the same name is used consistently across related objects. "SuperCollider in Terminal" could have come from anywhere, but "SuperCollider runInTerminal" (perhaps along with the first few characters of the executed string as well) makes the "cause" of the window unambiguously clear.

Reply:
I agree that `SuperCollider runInTerminal` is better. Incidentally, there doesn't seem to be a way to set the title in macOS terminal, or at least I can't figure it out (instead, it shows the filename of the script, which is `"SCStringForTerminal" ++ (this.hash.asString) ++ ".command";`).

Reply:
Changed to "SuperCollider runInTerminal"